### PR TITLE
feat(AddToCart) ICKY-324 Handled Product variation option and Add To Cart button

### PR DESCRIPTION
**Pull request by TriptiVermaXP**

### CODE REVIEW

File: pages/product/index.vue [hunk 1]
```diff
@@ -235,13 +236,22 @@ import { defineComponent, computed } from "@vue/composition-api"
...
+import {
+  useProductSSR,
+  useUiState,
+  usePurchaseLocation,
+  productGetters,
+  useCart,
+} from "@/composables"
 import { isFulfillmentOptionValid } from "@/composables/helpers"
+import { isProductVariationsSelected } from "~~/composables/helpers/validateProductVariations"
+import { AddItemsToCartParams } from "~~/composables/types/useCart"
```
Comment (kibo-chandradeeptalaha) [inline pages/product/index.vue]: Be consistent with the alias. Use @ instead of ~~

File: pages/product/index.vue [hunk 2]
```diff
@@ -292,10 +303,15 @@ export default defineComponent({
...
 
+    const isValidForAddToCart = ref(false)
     // Options section
     let shopperEnteredValues = []
 
-    const updateShopperEnteredValues = (attributeFQN, value, shopperEnteredValue) => {
+    const updateShopperEnteredValues = (
+      attributeFQN: string,
+      value: string,
+      shopperEnteredValue: undefined
```
Comment (kibo-chandradeeptalaha) [inline pages/product/index.vue]: Should this type be undefined ?

Comment (TriptiVermaXP) [reply pages/product/index.vue]: Resolved

File: pages/product/index.vue (hunk 1 above)
Comment (TriptiVermaXP) [reply pages/product/index.vue]: Done

File: components/KiboProductActions.vue [hunk 3]
```diff
@@ -87,6 +95,7 @@ export default defineComponent({
     }
 
     const addToCart = () => {
+      console.log("props", props.qty)
```
Comment (kibo-chandradeeptalaha) [inline components/KiboProductActions.vue:98]: Remove console.log

File: pages/product/index.vue [hunk 4]
```diff
@@ -336,28 +352,35 @@ export default defineComponent({
...
-        setFulfillment(selectedFulfillmentVal, purchaseLocation.value.code)
+        setFulfillment(selectedFulfillmentVal, shortName, purchaseLocation?.value?.code)
+        checkEligibilityForAddToCart()
       }
     }
 
+    const checkEligibilityForAddToCart = () => {
+      isValidForAddToCart.value =
+        isProductVariationsSelected(product.value) && Boolean(product.value.fulfillmentMethod)
+    }
```
Comment (kibo-kevinwatts) [inline pages/product/index.vue]: Convert `isValidForAddToCart` to a computed property
Remove `checkEligibilityForAddToCart` and its calls

File: pages/product/index.vue [hunk 5]
```diff
@@ -336,28 +352,35 @@ export default defineComponent({
...
-      const addToCartVariables = {
-        productToAdd: {
-          product: {
-            productCode: product.value?.productCode,
-          },
-          quantity: qtySelected,
-          fulfillmentMethod: product.value?.fulfillmentMethod,
-          purchaseLocation: product.value?.purchaseLocationCode,
+    const addToCart = async () => {
+      const productToAdd: CartItem = {
```
Comment (kibo-kevinwatts) [inline pages/product/index.vue]: Use the AddToCartInput helper I added to create the productToAdd variable

**Review by kibo-kevinwatts — CHANGES_REQUESTED** [review]

File: pages/product/index.vue (hunk 4 above)
Comment (TriptiVermaXP) [reply pages/product/index.vue]: done

File: pages/product/index.vue (hunk 5 above)
Comment (TriptiVermaXP) [reply pages/product/index.vue]: done

**Review by kibo-kevinwatts — APPROVED** [review]

**Pull request merged**